### DISCUSSION TITLE
How to do @$() which only breaks on newlines

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#4942] [Q&A]
Very often I have to do the follwoing

cmd2 @($(cmd1).splitlines())

where `cmd1` generates newline-separated strings (e.g. file paths) which `cmd2` will process.

If these individual strings did not contain spaces I could use `@$()`, but more often than not they do contain spaces.

Is there a shortcut like `@$()` which only breaks on newlines?

**Comment by anki-code** [comment]
Welcome to https://github.com/xonsh/xonsh/issues/3924

**Reply by halloleo** [reply]
Thanks @anki-code for the link

I glanced through the issue thread. It is rather long. As I understand the thread the current state in xonsh is:

1. There is no substitution operator for this yet.
2. One option is to **change the existing `@$` operator**. (_Problem_: Breaks backwards comaptibility _severly_.)
3. Another option is **a new operator for this**. (_Problem_: More operators make the language more confusing.)

Have I summarised this appropriately?

**Reply by anki-code** [reply]
>  It is rather long

In the nutshell my suggestion described here - https://github.com/anki-code/xonsh-operators-proposal/blob/main/XEP-2.rst

> Have I summarised this appropriately?

Mostly yes.
